processing/src/main/java/io/druid/query/extraction/ExtractionFn.java

```diff
@@ -29,7 +29,8 @@
     @JsonSubTypes.Type(name = "partial", value = MatchingDimExtractionFn.class),
     @JsonSubTypes.Type(name = "searchQuery", value = SearchQuerySpecDimExtractionFn.class),
     @JsonSubTypes.Type(name = "javascript", value = JavascriptExtractionFn.class),
-    @JsonSubTypes.Type(name = "timeFormat", value = TimeFormatExtractionFn.class)
+    @JsonSubTypes.Type(name = "timeFormat", value = TimeFormatExtractionFn.class),
+    @JsonSubTypes.Type(name = "explicitLookup", value = ExplicitDimRenameFn.class)
```

Comment:
It is essentially a double-key on the type name in that case, which could be solved with nested objects. then it might look something like this:


```json
"extractionFn":{
  "type":"lookup"
  "value":{
    "type":"map",
    "lookup":{"key1":"value1"},
    "retainMissingValues":true
  }
}
```

Does that capture it or did you have a different thought on the obj structure?